app/src/main/java/com/app/calculator/presenter/OperationPresenter.java

```diff
@@ -27,7 +27,10 @@ private void onButtonPressed() {
         else model.setSecondNumber(Integer.parseInt("0"));
 
         if(!view.getOperator().equalsIgnoreCase(Operator.DEF.getSymbol())) {
-            view.setResult(String.valueOf(model.getOperation(view.getOperator())));
+            if(view.getOperator().equalsIgnoreCase(Operator.DIV.getSymbol()) && view.getSecondNumber().equalsIgnoreCase("0")){
```

Comment:
view.getOperator() is being used a lot here, consider using a variable to avoid excesive access to the same method.